### PR TITLE
Bump flow to v0.85

### LINKED ISSUE #1270: Fix Flow v0.85 errors
Flow v0.85 introduced (or rather, unmissed) some type errors / missing annotations, as detailed in this blog post: https://medium.com/flow-type/asking-for-required-annotations-64d4f9c1edf8 

We currently have [10 new errors in our CI](https://circleci.com/gh/nusmodifications/nusmods/11222?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). It would be good to fix these sooner rather than later so that they don't block any future upgrades or snowball when new releases introduce more errors.

### PR DESCRIPTION
**Pull request by Blasz**
Closes #1270

### CODE REVIEW

File: www/flow-typed/package-dep-libdefs.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ declare module '@material/snackbar' {
   declare module.exports: any;
 }
 
+declare module 'bootstrap' {
```
Comment (ZhangYiJiang) [inline www/flow-typed/package-dep-libdefs.js]: Please revert the changes to this file. It's automatically generated by a script which I don't have time to fix, but in short this change is unwanted. Sorry!

File: www/src/js/views/components/KeyboardShortcuts.jsx [hunk 2]
```diff
@@ -139,13 +139,19 @@ export class KeyboardShortcutsComponent extends PureComponent<Props, State> {
...
 
-  renderShortcut = (shortcut: Shortcut) => {
+  renderSingleShortcut = (shortcut: string): Element<ElementType> => {
+    const capitalized = shortcut.replace(/\b([a-z])/, (c) => c.toUpperCase());
+    return <kbd key={shortcut}>{capitalized}</kbd>;
+  };
+
+  renderShortcut = (
+    shortcut: Shortcut,
+  ): Element<ElementType> | Array<Element<ElementType> | string> => {
```
Comment (ZhangYiJiang) [inline www/src/js/views/components/KeyboardShortcuts.jsx]: I'm not entirely sure splitting the functions is necessary, but render functions should be typed `Node` (imported from React), which encompasses everything React accepts from `render()`.

File: www/src/js/views/components/cors-stats/CorsRound.jsx [hunk 3]
```diff
@@ -72,7 +72,7 @@ function renderBidTable(groupedStats: GroupedBiddingStat[]) {
  * Detailed bidding info for a single round. We first group by faculty then group
  * by studentType.
  */
-function CorsRound(props: Props) {
+function CorsRound(props: Props): Element<ElementType> | Element<ElementType>[] {
```
Comment (ZhangYiJiang) [inline www/src/js/views/components/cors-stats/CorsRound.jsx]: As above, type this as `Node` instead

File: www/src/js/views/components/cors-stats/CorsSummary.jsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ const STUDENT_TYPE_LABELS = {
  * Summary statistics consists of minimum amount of bid points required
  * to get a module for each type of student in each faculty
  */
-export default function(props: Props) {
+export default function(props: Props): Element<ElementType>[] {
```
Comment (ZhangYiJiang) [inline www/src/js/views/components/cors-stats/CorsSummary.jsx]: And this too

File: www/src/js/views/components/KeyboardShortcuts.jsx (hunk 2 above)
Comment (Blasz) [reply www/src/js/views/components/KeyboardShortcuts.jsx]: Ah true, completely forgot about `Node`!

**Comment by Blasz** [comment]
I've addressed the feedback. Fixing the prettier violation broke flow types, it couldn't seem to handle the spacing (which I think is incorrect anyway, you may need to install/update a flow prettier plugin to disable this rule) so I changed it to use a return type.
